vpp-manager/uplink/common.go

```diff
@@ -55,6 +55,36 @@ func (d *UplinkDriverData) GetName() string {
 	return d.name
 }
 
+func (d *UplinkDriverData) removeLinuxIfConf(setIfDown bool) {
+	link, err := netlink.LinkByName(d.params.MainInterface)
+	if err != nil {
+		log.Errorf("Error finding link %s: %s", d.params.MainInterface, err)
+	} else {
+		// Remove routes to not have them conflict with vpptap0
+		for _, route := range d.conf.Routes {
+			log.Infof("deleting Route %s", route.String())
+			err := netlink.RouteDel(&route)
+			if err != nil {
+				log.Errorf("cannot delete route %+v: %+v", route, err)
+			}
+		}
+		// Remove addresses to not have them conflict with vpptap0
+		for _, addr := range d.conf.Addresses {
+			err := netlink.AddrDel(link, &addr)
+			if err != nil {
+				log.Errorf("Error adding address %s to tap interface : %+v", addr, err)
+			}
+		}
+
+		if d.conf.IsUp && setIfDown {
+			err = netlink.LinkSetDown(link)
+			if err != nil {
```

Comment:
In SafeSetInterfaceDownByName , if LinkSetDown errored we attempted to put the link back up, should we do it here as well?

Reply:
Right, yesterday I didn't recall why we did have it. But on a second thought it makes sense